morango/controller.py

```diff
@@ -25,10 +27,20 @@ def serialize_into_store(self):
         with transaction.atomic():
             defaults = {'instance_id': current_id.id, 'counter': current_id.counter}
 
+            new_store_records = []
+            new_rmc_records = []
+
+            # create Q objects for filtering by prefixes
+            if filters:
+                prefix_condition = functools.reduce(lambda x, y: x | y, [Q(_morango_partition__startswith=prefix) for prefix in filters])
+
             # filter through all models with the dirty bit turned on
             syncable_dict = _profile_models[self.profile]
             for (_, klass_model) in iteritems(syncable_dict):
-                for app_model in klass_model.objects.filter(_morango_dirty_bit=True):
+                klass_queryset = klass_model.objects.filter(_morango_dirty_bit=True)
+                if filters:
```

Comment:
Might be more semantic to conditionalize on `prefix_condition` here (might mean setting it to `None` by default at the top)